### DISCUSSION TITLE
Panic Error

### DISCUSSION THREAD

**Discussion by Tucubanito07** [dani-garcia/vaultwarden#4319] [Q&A]
I updated the app on Unraid and for whatever reason it keeps giving me this error. 

[2024-02-05 13:14:30.135][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2024-02-05 13:14:30.749][request][INFO] POST /identity/connect/token
[2024-02-05 13:14:30.751][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-02-05 13:14:30.883][request][INFO] GET /api/sync
**[2024-02-05 13:14:30.884][panic][ERROR] thread 'rocket-worker-thread' panicked at 'Error loading ciphers:** **DatabaseError(Unknown, "database disk image is malformed")': src/db/models/cipher.rs:686**
   0: vaultwarden::init_logging::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: std::panicking::begin_panic_handler::{{closure}}
   3: std::sys_common::backtrace::__rust_end_short_backtrace
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: core::result::unwrap_failed
   7: vaultwarden::db::models::cipher::Cipher::find_by_user::{{closure}}::{{closure}}
   8: vaultwarden::db::models::cipher::Cipher::find_by_user::{{closure}}
   9: vaultwarden::db::models::cipher::Cipher::find_by_user_visible::{{closure}}
  10: vaultwarden::api::core::ciphers::sync::{{closure}}
  11: vaultwarden::api::core::ciphers::sync::into_info::monomorphized_function::{{closure}}
  12: rocket::server::<impl rocket::rocket::Rocket<rocket::phase::Orbit>>::route::{{closure}}
  13: rocket::server::hyper_service_fn::{{closure}}::{{closure}}
  14: tokio::runtime::task::raw::poll
  15: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
  16: tokio::runtime::scheduler::multi_thread::worker::run
  17: tokio::runtime::task::raw::poll
  18: std::sys_common::backtrace::__rust_begin_short_backtrace
  19: core::ops::function::FnOnce::call_once{{vtable.shim}}
  20: std::sys::unix::thread::Thread::new::thread_start
  21: <unknown>
  22: clone

**[2024-02-05 13:14:30.884][response][INFO] (sync) GET /api/sync?<data..> => 500 Internal Server Error**
[2024-02-05 13:14:31.185][request][INFO] POST /identity/connect/token
[2024-02-05 13:14:31.187][response][INFO] (login) POST /identity/connect/token => 200 OK

I have logged out of the app on my iPhone and when logging back in, it does not display all my password. However, on my Edge extension it has all of my password. Can someone please help me fix this issue.

**Comment by BlackDex** [comment]
Looks like your database is corrupted.

**Reply by Tucubanito07** [reply]
Do you happen to know how I can fix the database?

**Reply by BlackDex** [reply]
I think you are using sqlite. So try https://sqlite.org/recovery.html


Or restore from backups.
